user-data.sh

```diff
@@ -26,23 +29,27 @@ INSTALLATION_PATH="/tmp/weka"
 mkdir -p $INSTALLATION_PATH
 
 # install ofed
-OFED_NAME=ofed-${ofed_version}
-if [[ "${install_ofed_url}" ]]; then
-  wget ${install_ofed_url} -O $INSTALLATION_PATH/$OFED_NAME.tgz
+if [[ "${skip_ofed_installation}" == false ]]; then
+  OFED_NAME=ofed-${ofed_version}
+  if [[ "${install_ofed_url}" ]]; then
+    wget ${install_ofed_url} -O $INSTALLATION_PATH/$OFED_NAME.tgz
+  else
+    wget http://content.mellanox.com/ofed/MLNX_OFED-${ofed_version}/MLNX_OFED_LINUX-${ofed_version}-ubuntu18.04-x86_64.tgz -O $INSTALLATION_PATH/$OFED_NAME.tgz
+  fi
+
+  tar xf $INSTALLATION_PATH/$OFED_NAME.tgz --directory $INSTALLATION_PATH --one-top-level=$OFED_NAME
+  cd $INSTALLATION_PATH/$OFED_NAME/*/
+  ./mlnxofedinstall --without-fw-update --add-kernel-support --force 2>&1 | tee /tmp/weka_ofed_installation
+  /etc/init.d/openibd restart
+
+  disk=$(lsblk -o NAME,HCTL,SIZE,MOUNTPOINT | grep "3:0:0:0" | awk '{print $1}')
+  apt update -y
+  apt install -y jq
 else
-  wget http://content.mellanox.com/ofed/MLNX_OFED-${ofed_version}/MLNX_OFED_LINUX-${ofed_version}-ubuntu18.04-x86_64.tgz -O $INSTALLATION_PATH/$OFED_NAME.tgz
+  disk=$(lsblk -o NAME,HCTL,SIZE,MOUNTPOINT | grep "1:0:0:0" | awk '{print $1}')
```

Comment:
no need for else, just put this line out of the if instead of duplicating it.

Reply:
you do this line in both cases, it should be out of if, this is why no need for else